File: submissions/424-longest-repeating-character-replacement.py
class Solution(object):
    def characterReplacement(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        count = {}
        max_freq = 0
        max_len = 0
        left = 0
        for right in range(len(s)):
            count[s[right]] = count.get(s[right], 0) + 1
            max_freq = max(max_freq, count[s[right]])
            window = right - left + 1
            if window - max_freq > k:
                count[s[left]] -= 1
                left += 1
            max_len = max(max_len, right - left + 1)
        return max_len
        

File: submissions/424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int n = s.size();
        int result = 0;
        int i = 0;
        int mostF = 0;
        int count[26] = {0};
        for(int j = 0; j < n; ++j) {
            count[s[j]- 'A']++;
            mostF = max(mostF, count[s[j]- 'A']);
            while((j - i + 1) - mostF > k) {
                count[s[i]- 'A']--;
                i++;
            }
            result = max(result, j - i + 1);
        }
        return result;
    }
};
